tests/test_function.py

```diff
@@ -301,19 +300,16 @@ def f(x: int):
 
     # At this point, there should be nothing to register, so a call should not clear
     # the cache.
-    assert f._pending == []
-    f._resolve_pending_registrations()
-    assert len(f._cache) == 1
+    assert f._resolver
+    assert len(f._methods_registry._resolver._cache) == 1
 
     @f.dispatch
     def f(x: str):
         pass
 
     # Now there is something to register. A call should clear the cache.
-    assert len(f._pending) == 1
-    f._resolve_pending_registrations()
-    assert len(f._pending) == 0
-    assert len(f._cache) == 0
+    f._resolver
+    assert f._methods_registry._resolver._cache == {}
```

Comment:
```suggestion
    assert not f._methods_registry._resolver._cache
```

(slightly more Pythonic)